core/include/cubos/core/reflection/traits/inherits.hpp

```diff
@@ -0,0 +1,41 @@
+#include <string>
+#include <unordered_set>
+
+#include <cubos/core/reflection/type.hpp>
+
+namespace cubos::core::reflection
+{
+    /// @brief Provides inheritance relationship between types.
+    /// @see See @ref examples-core-reflection-traits-inherits for an example of using this trait.
+    /// @ingroup core-reflection
+    class InheritsTrait
+    {
+    public:
+        ~InheritsTrait() = default;
+
+        /// @brief Constructs.
+        InheritsTrait() = default;
```

Comment:
I would make the constructor receive a `const Type&`, as it never makes sense to add this trait without setting a type.